core/arch/clean_call_opt_shared.c

```diff
@@ -732,14 +732,11 @@ analyze_clean_call(dcontext_t *dcontext, clean_call_info_t *cci, instr_t *where,
     }
 # ifdef X86
     /* 9. derived fields */
-    /* XXX: For x64, skipping a single reg or flags still results in a huge
-     * code sequence to put in place: we may want to still use out-of-line
-     * unless multiple regs are able to be skipped.
+    /* Use out-of-line calls if more than 3 SIMD or GP registers have to be saved.
      * XXX: This should probably be in arch-specific clean_call_opt.c.
      */
-    if ((cci->num_simd_skip == 0 /* save all xmms */ &&
-         cci->num_regs_skip == 0 /* save all regs */ &&
-         !cci->skip_save_flags) ||
+    if ((NUM_SIMD_REGS - cci->num_simd_skip) > 3  /* save more than 3 xmms */ ||
```

Comment:
Not to the value, though I always prefer named constants, even if local, instead of magic literals, with a comment explaining whether this value was arrived at through experimentation or it's just a guess.